packages/headless/src/api/search/search/search-response.ts

```diff
@@ -13,6 +14,7 @@ import {SecurityIdentity} from './security-identity';
 import {PhrasesToHighlight, TermsToHighlight} from './stemming';
 
 export interface SearchResponseSuccess {
+  generateAutomaticFacets: AutomaticFacets;
```

Comment:
Double check the optionality of the value in the response.

I think `generatedAutomaticFacets` won't be returned by the API if none are requested ? And if so, it should be `generateAutomaticFacets?: AutomaticFacets`

It's very important that all the types defined here match very well how the API behaves, otherwise it will start creeping up as bug in a lot of different places.

Reply:
I made it optional to match how the API behaves